pkg/cmd/adm/restart_test.go

```diff
@@ -1,216 +1,206 @@
 	"k8s.io/apimachinery/pkg/types"
-	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/rest/fake"
+	cgtesting "k8s.io/client-go/testing"
+	cmdtesting "k8s.io/kubectl/pkg/cmd/testing"
+	"k8s.io/kubectl/pkg/scheme"
 )
 
 func TestRestartDeployment(t *testing.T) {
 	// given
-	SetFileConfig(t, Host(), Member())
-
-	for _, clusterName := range []string{"host", "member1"} {
-		clusterType := configuration.Host
-		if clusterName != "host" {
-			clusterType = configuration.Member
-		}
-		namespace := fmt.Sprintf("toolchain-%s-operator", clusterType)
-		namespacedName := types.NamespacedName{
-			Namespace: namespace,
-			Name:      "cool-deployment",
-		}
-		term := NewFakeTerminalWithResponse("Y")
-
-		t.Run("restart is successful for "+clusterName, func(t *testing.T) {
-			// given
-			deployment := newDeployment(namespacedName, 3)
-			newClient, fakeClient := NewFakeClients(t, deployment)
-			numberOfUpdateCalls := 0
-			fakeClient.MockUpdate = requireDeploymentBeingUpdated(t, fakeClient, namespacedName, 3, &numberOfUpdateCalls)
-			ctx := clicontext.NewCommandContext(term, newClient)
-
-			// when
-			err := restart(ctx, clusterName, "cool-deployment")
-
-			// then
-			require.NoError(t, err)
-			AssertDeploymentHasReplicas(t, fakeClient, namespacedName, 3)
-			assert.Equal(t, 2, numberOfUpdateCalls)
-		})
-
-		t.Run("list deployments when no deployment name is provided for "+clusterName, func(t *testing.T) {
-			// given
-			deployment := newDeployment(namespacedName, 3)
-			newClient, fakeClient := NewFakeClients(t, deployment)
-			numberOfUpdateCalls := 0
-			fakeClient.MockUpdate = requireDeploymentBeingUpdated(t, fakeClient, namespacedName, 3, &numberOfUpdateCalls)
-			term := NewFakeTerminalWithResponse("Y")
-			ctx := clicontext.NewCommandContext(term, newClient)
-
-			// when
-			err := restart(ctx, clusterName)
-
-			// then
-			require.EqualError(t, err, "at least one deployment name is required, include one or more of the above deployments to restart")
-			AssertDeploymentHasReplicas(t, fakeClient, namespacedName, 3)
-			assert.Equal(t, 0, numberOfUpdateCalls)
-			assert.Contains(t, term.Output(), fmt.Sprintf("Existing deployments in toolchain-%s-operator namespace", clusterType))
-			assert.Contains(t, term.Output(), "cool-deployment")
-		})
-
-		t.Run("restart fails - cannot get the deployment for "+clusterName, func(t *testing.T) {
-			// given
-			deployment := newDeployment(namespacedName, 3)
-			newClient, fakeClient := NewFakeClients(t, deployment)
-			numberOfUpdateCalls := 0
-			fakeClient.MockUpdate = requireDeploymentBeingUpdated(t, fakeClient, namespacedName, 3, &numberOfUpdateCalls)
-			fakeClient.MockGet = func(ctx context.Context, key runtimeclient.ObjectKey, obj runtimeclient.Object, opts ...runtimeclient.GetOption) error {
-				return fmt.Errorf("some error")
+	tests := map[string]struct {
```

Comment:
If it's not too tricky , I think we could add some tests/scenarios also for:
- there's some other noise pods in the namespace that should not be restarted
- the pod/deployment doesn't restart correctly ( both olm and non-olm )